### ISSUE TITLE
Abort trigger aborts, but the file upload form resumes uploading from scratch.

### ISSUE THREAD

**Issue by derpycoder** [bigskysoftware/htmx#2756]
I have copied pasted the example code from HTMX documentation, that depicts file upload:

```html
<form id="form" hx-post="/upload" enctype="multipart/form-data">
    <input type="file" name="file" id="file"/>
    <button>Upload</button>
    <progress id="progress" value="0" max="100"></progress>
    <button onclick="htmx.trigger('#form', 'htmx:abort')">Cancel Request</button>
</form>
```

I verified that the abort event is triggered, and visually the file upload gets aborted, followed by these errors:
![image](https://github.com/user-attachments/assets/4b684420-0e91-44d2-a34e-3ebd8fc09c6c)

And Immediately file upload resumes, resulting in file submission.

I am using HTMX@2.0.1, with Go & Templ.

**Comment by MichaelWest22** [comment]
I don't think this is a bug.  There are two submit buttons in your form.  one just uploads the file and the second has a click handler to cancel the current action and then submits the form again because it is just a normal submit button.  You would need to preventDefault() in your onClick or maybe set the type of your button with `type="button"` which will remove the default submit behavior.
